src/containerapp/azext_containerapp/custom.py

```diff
@@ -2413,3 +2468,276 @@ def delete_environment_telemetry_app_insights(cmd,
+    try:
+        containerapp_env_def = containerapp_env_telemetry_app_insights_decorator.show()
+    except:
+        pass
+
+    if not containerapp_env_def:
+        raise ResourceNotFoundError("The containerapp environment '{}' does not exist".format(name))
+
+    try:
+        r = containerapp_env_def["properties"]["appInsightsConfiguration"]
+        if r is None:
+            raise ValidationError("The containerapp environment '{}' does not have app insights enabled.".format(name)) from e
+        
+        r = {}
+
+        if "connectionString" in containerapp_env_def["properties"]["appInsightsConfiguration"]:
+            safe_set(r, "appInsightsConfiguration", "connectionString", value=DEFAULT_CONFIGURED_STR)
+
+        enable_open_telemetry_traces = False
+        existing_traces = safe_get(containerapp_env_def, "properties", "openTelemetryConfiguration", "tracesConfiguration", "destinations")
+        if existing_traces and APP_INSIGHTS_DEST in existing_traces:
+            enable_open_telemetry_traces = True
+        
+        safe_set(r, "appInsightsConfiguration", "enable-open-telemetry-traces", value=enable_open_telemetry_traces)
+
+        enable_open_telemetry_logs = False
+        existing_logs = safe_get(containerapp_env_def, "properties", "openTelemetryConfiguration", "logsConfiguration", "destinations")
+        if existing_logs and APP_INSIGHTS_DEST in existing_logs:
+            enable_open_telemetry_logs = True
+        
+        safe_set(r, "appInsightsConfiguration", "enable-open-telemetry-logs", value=enable_open_telemetry_logs)
+
+        return r
+    except Exception as e:
+        raise ValidationError("The containerapp environment '{}' does not have app insights enabled.".format(name)) from e
+
+
+def add_environment_telemetry_otlp(cmd,
+                                   name,
+                                   resource_group_name,
+                                   otlp_name,
+                                   endpoint,
+                                   insecure=False,
+                                   enable_open_telemetry_traces=False,
+                                   enable_open_telemetry_logs=False,
+                                   enable_open_telemetry_metrics=False,
+                                   headers=None,
+                                   no_wait=False):
+    raw_parameters = locals()
+    containerapp_env_telemetry_otlp_decorator = ContainerappEnvTelemetryOtlpPreviewSetDecorator(
+        cmd=cmd,
+        client=ManagedEnvironmentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    containerapp_env_telemetry_otlp_decorator.register_provider(CONTAINER_APPS_RP)
+
+    try:
+        r = containerapp_env_telemetry_otlp_decorator.show()
+    except CLIError as e:
+        handle_raw_exception(e)
+
+    existing_otlps = safe_get(r, "properties", "openTelemetryConfiguration", "destinationsConfiguration", "otlpConfigurations")
```

Comment:
There is a warning relate to `r`:
`Local variable 'r' might be referenced before assignment `